addons/account/models/account_move.py

```diff
@@ -4935,6 +4944,41 @@ def _post(self, soft=True):
         # reconcile if state is in draft and move has reversal_entry_id set
         draft_reverse_moves = to_post.filtered(lambda move: move.reversed_entry_id and move.reversed_entry_id.state == 'posted')
 
+        # deal with the eventually related draft moves to the ones we want to post
+        partials_to_unlink = self.env['account.partial.reconcile']
+
+        for move in self:
+            partial_vals, dummy = move._get_reconciled_invoices_partials()
+            for (partial, dummy, counterpart_move_line) in partial_vals:
+                counterpart_move = counterpart_move_line.move_id
+
+        for aml in self.line_ids:
+            for partials, counterpart_field in [(aml.matched_debit_ids, 'debit_move_id'), (aml.matched_credit_ids, 'credit_move_id')]:
+                for partial in partials:
+                    counterpart_move =  partial[counterpart_field].move_id
+                    if counterpart_move.state == 'posted' or counterpart_move in to_post:
+                        if partial._get_draft_exchange_move_vals() != partial.draft_exchange_move_vals:
+                            if partial.exchange_move_id:
+                                # draft invoice changed since it was reconciled, the exchange diff entry isn't corect anymore
+                                # and the user has to re-do the reconciliation. Existing draft exchange move will be unlinked
+                                partials_to_unlink |= partial
+
+                        elif partial.exchange_move_id:
+                            to_post |= partial.exchange_move_id
```

Comment:
```py
if partial.exchange_move_id:
    if partial._get_draft_exchange_move_vals() == partial.draft_exchange_move_vals:
        # Post the entry
        ...
    else:
        # Delete the partial.
        # TODO: add a comment explaining the move will be unlinked/cancelled by the override of "unlink" in the partial
```

Reply:
see my other comment: we can't rely on partial.exchange_move_id existing